### PR TITLE
[6.5 Feature] Content View Export Import - Part 2 - automated

### PR DESCRIPTION
**Pull request by jyejare**
Part 2 is here:

- 7 more tests are automated

### CODE REVIEW

**Comment by jyejare** [comment]
The Test Results:

```

# py.test tests/foreman/cli/test_satellitesync.py::ContentViewSync
========================================== test session starts ==========================================
platform linux -- Python 3.6.8, pytest-4.0.2, py-1.7.0, pluggy-0.8.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/jitendrayejare/Desktop/RedHat/RoboTelloNew/robottelo, inifile:
plugins: xdist-1.26.1, services-1.3.1, mock-1.10.0, forked-1.0.2, cov-2.6.1
collecting ... 2019-02-19 13:44:03 - conftest - DEBUG - Collected 59 test cases

collected 59 items / 43 deselected

tests/foreman/cli/test_satellitesync.py .                                                         [100%]

===================== 14 passed, 2 skipped in 1868.87 seconds =====================
```

File: tests/foreman/cli/test_satellitesync.py [hunk 1]
```diff
@@ -439,17 +479,21 @@ def test_positive_export_import_redhat_cv(self):
...
         })
         exported_tar = '{0}/export-{1}-{2}.tar'.format(
-            self.export_base, self.exporting_cv_name, self.exporting_cvv_id)
+            self.export_base, rhel_cv_name, exporting_cvv_id)
         result = ssh.command("[ -f {0} ]".format(exported_tar))
         self.assertEqual(result.return_code, 0)
         exported_packages = Package.list({'content-view-version-id': exporting_cvv_id})
         self.assertTrue(len(exported_packages) > 0)
-        imp_rhel_repo = ContentViewSync._enable_rhel_content(self.importing_org, sync=False)
+        Subscription.delete_manifest({
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_satellitesync.py]: Was that extra space here intended?

File: tests/foreman/cli/test_satellitesync.py [hunk 2]
```diff
@@ -763,6 +807,382 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
+        :steps:
+
+            1. Create product and mirror-on-sync repository with custom contents
+            2. Sync the repository
+            3. Create CV with above product and publish
+            4. Attempt to export CV version contents to a directory
+
+        :expectedresults:
+
+            1. Export fails with error 'All exported repositories must be set to an immediate
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_satellitesync.py]: please change the expected result here which is not similar to the assertion  message in the test

File: tests/foreman/cli/test_satellitesync.py [hunk 3]
```diff
@@ -763,6 +807,382 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
+        """Importing CV version having mirror-on-sync repo throws error
+
+        :id: dfa0cd5f-1596-4097-b505-06bc14de51dd
+
+        :steps:
+
+            1. Create product and mirror-on-sync repository with custom contents
+            2. Sync the repository
+            3. Create CV with above product and publish
+            4. Attempt to export CV version contents to a directory
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_satellitesync.py]: are you trying to import or export here?

File: tests/foreman/cli/test_satellitesync.py [hunk 4]
```diff
@@ -763,6 +807,382 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
+            str(metadata.get('minor')), exporting_cv['versions'][0]['version'].split('.')[1])
+        self.assertIsNotNone(metadata.get('repositories'))
+        cvv_repository = metadata['repositories'][0]
+        self.assertEqual(cvv_repository.get('content_type'), 'yum')
+        self.assertEqual(cvv_repository.get('label'), self.exporting_repo_name)
+        self.assertIsNotNone(cvv_repository.get('backend_identifier'))
+        self.assertIsNotNone(cvv_repository.get('on_disk_path'))
+        self.assertIsNotNone(cvv_repository.get('relative_path'))
+        self.assertIsNotNone(
+            exporting_repo['content-counts']['packages'],
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_satellitesync.py]: assertIsNotNone is a valid assertion?

File: tests/foreman/cli/test_satellitesync.py [hunk 5]
```diff
@@ -763,6 +807,382 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
+            len(cvv_repository.get('rpm_filenames')))
+        self.assertIsNotNone(
+            exporting_repo['content-counts']['errata'],
```
Comment (omkarkhatavkar) [inline tests/foreman/cli/test_satellitesync.py]: assertIsNotNone is a valid assertion here?

File: tests/foreman/cli/test_satellitesync.py (hunk 1 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py]: Nop, but Git showing it as extra space, it not in actual code. Btw, I had everything now on 1 line.

File: tests/foreman/cli/test_satellitesync.py (hunk 3 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py]: Export.

We have to export it to the directory only. And we can import from directory only. In both export and import, directory comes into the pic.

File: tests/foreman/cli/test_satellitesync.py (hunk 2 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py]: Yeah, nice catch. Thanks, updated.

File: tests/foreman/cli/test_satellitesync.py (hunk 4 above)
Comment (jyejare) [reply tests/foreman/cli/test_satellitesync.py]: Again a nice catch, Updated ! :)

**Comment by omkarkhatavkar** [comment]
ACK

File: tests/foreman/cli/test_satellitesync.py [hunk 6]
```diff
@@ -763,6 +805,383 @@ def test_negative_export_cv_with_on_demand_repo(self):
...
+
+            1. Create product and mirror-on-sync repository with custom contents
+            2. Sync the repository
+            3. Create CV with above product and publish
+            4. Attempt to export CV version contents to a directory
+
+        :expectedresults:
+
+            1. Export fails with error 'The Repository '<repo_name>' is set with Mirror-on-Sync '
+             'to YES. Please change Mirror-on-Sync to NO and try the import again' should be
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_satellitesync.py]: This is where make-docstrings fails
```* Enumerated list ends without a blank line; unexpected unindent.```

**Pull request merged**